tests/main/snap-run-gdbserver/task.yaml

```diff
@@ -0,0 +1,38 @@
+summary: Check that `snap run --gdbserver` works
+
+# testing only here to avoid having to install gdb/gdbserver everywhere
+systems: [ubuntu-16.04-*, ubuntu-18.04-*, ubuntu-2*]
+
+prepare: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    install_local test-snapd-sh
+    apt install -y gdbserver gdb
+
+restore: |
+    apt autoremove -y gdbserver gdb
+
+execute: |
+  echo "Test snap run --gdbserver works"
+
+  # run the gdbserver command as a user
+  su -l -c 'snap run --gdbserver test-snapd-sh.sh -c "echo hello-hello-hello uid:$UID"' test >stdout &
```

Comment:
Please use session-tool and run this as with `systemd-run` to avoid leaking any processes.